src/popup.ts

```diff
@@ -0,0 +1,55 @@
+import { DatumNode } from "./types";
+
+export default function handlePopup(p: DatumNode) {
+  const popup = document.getElementById("diseasePopup");
+  const sunburst = document.getElementById("sunburst"); // Reference to the sunburst container
+
+  if (!popup || !sunburst) {
+    console.error("Required elements not found.");
+    return;
+  }
+
+  // Display the popup for disease nodes
+  const detailsHtml =
+    p.children
+      ?.map((child) => {
+        // Check if this child contains a 'Link'
+        if (child.data.name.startsWith("Link:")) {
+          // Extract the URL from the text, assuming it's the part after the colon
+          const urlMatch = child.data.name.match(/Link:\s*(.*)/);
+          const url = urlMatch ? urlMatch[1] : "";
+          // Format the URL as a clickable link, but keep 'Link:' as plain text
+          return `<strong>Link:</strong> <a href="${url}" target="_blank" rel="noopener noreferrer" style="color: purple;">${url}</a>`;
+        }
+        // Format other details, making the part before ':' bold
+        return child.data.name.replace(/(^[^:]+):/, "<strong>$1:</strong>");
+      })
+      .join("<br>") ?? "No details available";
```

Comment:
Works for now but may be easier to have text elements already templated out and we simply fill in the text content rather than rebuilding them with stringified HTML each time?